src/flash.test.ts

```diff
@@ -0,0 +1,286 @@
+import { test } from 'tap'
+import Fastify from 'fastify'
+import request from 'request'
+import fastifyCookie from 'fastify-cookie'
+import fastifySession from 'fastify-session'
+import { AddressInfo } from 'net'
+
+import fastifyFlash from '.'
+import { ExtendedRequest, ExtendedReply } from './flash'
+
+test('should set error message and and clear up after displaying.', t => {
+  t.plan(8)
+  const fastify = Fastify()
+
+  const options = {
+    secret: 'cNaoPYAwF60HZJzkcNaoPYAwF60HZJzk',
+  }
+
+  fastify.register(fastifyCookie)
+  fastify.register(fastifySession, options)
+  fastify.register(fastifyFlash)
+
+  fastify.get('/test', (req: ExtendedRequest, reply: ExtendedReply) => {
+    const count = req.flash('error', 'Something went wrong')
+
+    t.equal(count, 1)
+    t.equal(Object.keys(req.session.flash).length, 1)
+    t.equal(req.session.flash['error'].length, 1)
+    const error = reply.flash('error')
+    t.equal(Object.keys(req.session.flash).length, 0)
+    reply.send({ error })
+  })
+  fastify.listen(0, err => {
+    fastify.server.unref()
+    t.error(err)
+    request(
```

Comment:
I would recommend to use `fastify.inject()` instead, tests will be faster.

Reply:
OK, I'll explore the API ASAP.